### PR TITLE
Add ARM DesignStart FPGA V2C-DAPLink shield support

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add support for the ARM DesignStart FPGA V2C-DAPLink shield.

### CODE REVIEW

**Comment by henrikbrixandersen** [comment]
Adding this to the v2.4.0 milestone in the hope we can ship this release with "full" ARM DesignStart FPGA reference design support. This shield is an official companion shield from ARM Ltd. to aid in getting started with ARM DesignStart FPGA.

Using the shield requires boot-up configuration of the QSPI MUX in some configurations, so the shield is not easily used out-of-tree without the `board.c` logic introduced here.

File: boards/arm/arty/board.c [hunk 1]
```diff
@@ -58,9 +58,9 @@ int board_daplink_qspi_mux_select(enum board_daplink_qspi_mux_mode mode)
 bool board_daplink_is_fitted(void)
 {
 	/*
-	 * The DAPLINK_fitted_n signal is routed an IRQ line. It is used as a
-	 * level-detect non-interrupt signal to determine if the DAPLink
-	 * shield is fitted.
+	 * The DAPLINK_fitted_n signal is routed to an IRQ line. It is used as a
+	 * level-detect non-interrupt signal to determine if the DAPLink shield
+	 * is fitted.
```
Comment (erwango) [inline boards/arm/arty/board.c]: This should be moved to first commit

File: boards/arm/arty/dts/bindings/arm,daplink-qspi-mux.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ properties:
       required: true
       description: |
         IRQ line connected to the level-detect non-interrupt DAPLink shield
-        presence signal
+        fitted signal
```
Comment (erwango) [inline boards/arm/arty/dts/bindings/arm,daplink-qspi-mux.yaml]: Ditto

**Review by erwango — CHANGES_REQUESTED** [review]
Minor comment.
Otherwise, from shield point of view, this looks correct.

File: boards/arm/arty/board.c (hunk 1 above)
Comment (henrikbrixandersen) [reply boards/arm/arty/board.c]: @erwango Thank you for reviewing. Good catch! I must have performed a fixup on the wrong commit at some point. Fixed.

File: boards/arm/arty/dts/bindings/arm,daplink-qspi-mux.yaml (hunk 2 above)
Comment (henrikbrixandersen) [reply boards/arm/arty/dts/bindings/arm,daplink-qspi-mux.yaml]: Fixed, thanks.

**Review by erwango — APPROVED** [review]
LGTM from shield side

**Comment by henrikbrixandersen** [comment]
Rebased on master after #27888 and #24873 were merged. Refactored shield dts files a bit to allow for Cortex-M3 reference design support, added Kconfig for setting the initialization level for the board init function (due to #27985).

**Review by nashif — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@jfischer-phytec-iot could you take a look?

**Review by jfischer-no — APPROVED** [review]

**Review by pabigot — APPROVED** [review]

**Pull request merged**